Feature 212 device setup logic

htela05Xamk opened the pull request:
Resolves #212

htela05Xamk committed:
Added device setup logic

Logic to setup a device and configured an OkHttpClient to trust all certificates

htela05Xamk committed:
Added device setup logic

Logic to setup a device and configured an OkHttpClient to trust all certificates

htela05Xamk committed:
Added device setup logic

Logic to setup a device and configured an OkHttpClient to trust all certificates

anttileppa commented on app/src/main/java/fi/metatavu/muisti/exhibitionui/ExhibitionUIApplication.kt:
```diff
@@ -245,6 +208,40 @@ class ExhibitionUIApplication : Application() {
+     */
+    private fun getDeviceImageLoadStrategy(settingValue: String?): DeviceImageLoadStrategy{
+        settingValue ?: return DeviceImageLoadStrategy.mEMORY
+        return DeviceImageLoadStrategy.values().find { it.value.equals(settingValue, ignoreCase = true) }
```
Is there a reason why this method looks up the DeviceImageLoadStrategy as case insensitive and the getScreenOrientation uses just valueOf?

anttileppa commented on app/src/main/java/fi/metatavu/muisti/exhibitionui/views/SetupActivity.kt:
```diff
@@ -1,33 +1,183 @@
+     * Hides setup UI
+     */
+    private fun hideSetupUI() {
+
```
```suggestion
```

anttileppa commented on app/src/main/java/fi/metatavu/muisti/exhibitionui/views/SetupActivity.kt:
```diff
@@ -1,33 +1,183 @@
+        saveSetupButton.visibility = View.GONE
+        deviceNameLabel.visibility = View.GONE
+        deviceDescriptionLabel.visibility = View.GONE
+
```
```suggestion
```

anttileppa commented on app/src/main/java/fi/metatavu/muisti/exhibitionui/views/SetupViewModel.kt:
```diff
@@ -40,13 +64,43 @@ class SetupViewModel(application: Application): AndroidViewModel(application) {
+        try {
+            val devicesApi = MuistiApiFactory.getDevicesApi(deviceKey)
+            return devicesApi.getDeviceKey(deviceId = deviceId)
+        } catch(e:ClientException) {
```
```suggestion
        } catch (e: ClientException) {
```

htela05Xamk committed:
Apply suggestions from code review

Co-authored-by: Antti Leppä <antti.leppa@metatavu.fi>

htela05Xamk committed:
Added device setup logic

Logic to setup a device and configured an OkHttpClient to trust all certificates
